src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/ObjectWriter/Dwarf/DwarfInfo.cs

```diff
@@ -531,11 +530,16 @@ private static void DumpVarLocation(DwarfExpressionBuilder e, VarLoc loc)
                     e.OpPiece();
                     break;
                 case VarLocType.VLT_STK_REG:
-                    e.OpBReg(loc.C, loc.D);
+                    e.OpBReg(loc.B, loc.C);
                     e.OpPiece();
-                    e.OpReg(loc.B);
+                    e.OpReg(loc.D);
                     e.OpPiece();
                     break;
+                case VarLocType.VLT_FPSTK:
+                    // On ARM32 this is used to encode actual registers. This may be wrong for x86
+                    // which we don't support anyway.
+                    e.OpReg(loc.B);
```

Comment:
I would expect this to be correct on x86 as well - the Jit never places anything but the return value on the FPU stack.